sqlglot/dialects/dialect.py

```diff
@@ -277,6 +277,14 @@ def inline_array_sql(self: Generator, expression: exp.Array) -> str:
     return f"[{self.expressions(expression)}]"
 
 
+def safeconcat_sql(self: Generator, expression: exp.SafeConcat) -> str:
+    return self.func("CONCAT", *[exp.cast(e, "text") for e in expression.expressions])
+
+
+def safedpipe_sql(self: Generator, expression: exp.SafeDPipe) -> str:
+    return self.func("CONCAT", *[exp.cast(e, "text") for e in expression.flatten()])
```

Comment:
`CONCAT` is equivalent to `||` for presto, trino, clickhouse which are the dialects of interest that need this, so I transpiled the latter to the former because it was easier compared to flattening the `||`.